src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,115 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map.Entry;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {        
+        String testNgXml = getFileContent("./src/test/testng.xml");
+        HashMap<String, String> testFiles = getTestFiles(testNgXml); // <class name, package name>
+        
+        testFiles = excludeFiles(testFiles);
+        
+        for (Entry<String, String> testFileName : testFiles.entrySet()) {
+            assertTrue(isTestFileIncluded(testNgXml, testFileName));
+        }
+    }
+    
+    private String getFileContent(String file) throws FileNotFoundException {
+        return FileHelper.readFile(file);
+    }
+    
+    /**
+     * Files to be checked in testng.xml are added to testFiles
+     * 
+     * @param testNgXml    Contents of testng.xml
+     * @return             HashMap containing <class name, package name>
+     */
+    private HashMap<String, String> getTestFiles(String testNgXml) {       
+        // BaseComponentTestCase, BaseTestCase (files in current directory) excluded because 
+        // base classes are extended by the actual tests
+        
+        return addFilesToTestsRecursively(new HashMap<String, String>(), 
+                                              "./src/test/java/teammates/test/cases", true, "", testNgXml); 
+    }
+    
+    /**
+     * Exclude files which do not have tests
+     * 
+     * @param testFiles    Files to be checked before excluding tests
+     * @return             Files to be checked after excluding tests
+     */
+    private HashMap<String, String> excludeFiles(HashMap<String, String> testFiles) {
+        // Base*TestCase are base classes to be extended by the actual tests
+        testFiles.remove("BaseUiTestCase");
+        
+        // FeedbackQuestionUiTest is the base class for all Feedback*QuestionUiTest (different question types)
+        testFiles.remove("FeedbackQuestionUiTest");
+        
+        // Needs to be run only when changes are made to GodMode
+        testFiles.remove("GodModeTest");
+        
+        return testFiles;
+    }
+    
+    private boolean isTestFileIncluded(String testNgXml, Entry<String, String> testFileName) {
+        return testNgXml.contains("<class name=\"teammates.test.cases" 
+                                       + testFileName.getValue() + "." + testFileName.getKey() + "\" />");
+    }
+
+    /**
+     * Recursively add files from testng.xml which are to be checked
+     * 
+     * @param testFiles                       HashMap containing <class name, package name>
```

Comment:
I've studied the method and I can tell for sure that you don't. All the method does to the map in each recursion is adding things, which can be achieved also with `Map.putAll(...)`.